services/QuillLMS/app/queries/prompt_feedback_history.rb

```diff
@@ -30,17 +39,22 @@ def self.prompt_health_query(activity_id:, start_date: nil, end_date: nil)
     query
   end
 
+  # rubocop:disable Metrics/CyclomaticComplexity
```

Comment:
Rather than disable the cop, you could refactor this out:

```    
  def self.serialize_results(results)
    results.to_h { |result| [result.prompt_id, ResultSerializer.new(result).run] }
  end

  class ResultSerializer
    SERIALIZED_ATTRIBUTES = [
      :avg_attempts,
      :avg_confidence,
      :avg_time_spent,
      :display_name,
      :num_final_attempt_not_optimal,
      :num_final_attempt_optimal,
      :num_first_attempt_not_optimal,
      :num_first_attempt_optimal,
      :prompt_id,
      :session_count,
      :total_responses
    ]

    attr_reader :payload, :result

    def initialize(result)
      @result = result
      @payload = result.serializable_hash(only: SERIALIZED_ATTRIBUTES, include: [])
    end

    def run
      payload.merge(
        'avg_attempts' => avg_attempts,
        'avg_confidence' => avg_confidence,
        'avg_time_spent' => avg_time_spent,
        'num_sessions_with_consecutive_repeated_rule' => num_sessions_with_consecutive_repeated_rule,
        'num_sessions_with_non_consecutive_repeated_rule' => num_sessions_with_non_consecutive_repeated_rule
      )
    end

    def avg_attempts
      payload['avg_attempts']&.round(2)&.to_f || 0
    end

    def avg_confidence
      payload['avg_confidence'] ? payload['avg_confidence'].round(2) * 100 : 0
    end

    def avg_time_spent
      payload['avg_time_spent'] ? Utils::Numeric.seconds_to_human_readable_time(payload['avg_time_spent']) : 0
    end

    def num_sessions_with_consecutive_repeated_rule
      result.num_sessions_consecutive_repeated
    end

    def num_sessions_with_non_consecutive_repeated_rule
      result.num_sessions_non_consecutive_repeated
    end
  end
```